src/checkstyle/checks.xml

```diff
@@ -186,7 +186,8 @@
                 org.weakref.jmx.internal,
                 jersey.repackaged,
                 jdk.nashorn.internal,
-                jdk.internal" />
+                jdk.internal,
```

Comment:
offer an explaination when the check fails, similar to https://github.com/Teradata/presto/blob/e450a1a2c9fbecb48a5389d7c17b804b35cbb341/src/checkstyle/checks.xml#L74

Reply:
Even better, express it as a `modernizer` check:
https://github.com/Teradata/presto/blob/e450a1a2c9fbecb48a5389d7c17b804b35cbb341/src/modernizer/violations.xml

Reply:
This check is for banned packages, but we are banning a class. I think we need to use "illegalClasses"

Reply:
But I agree that modernizer is better

Reply:
Can one of you please explain why the modernizer check didn't fail for this:

https://github.com/prestodb/presto/pull/8867/files/e450a1a2c9fbecb48a5389d7c17b804b35cbb341#diff-72a17e84933869700df34870a3c6eaafR53

It seems likely that [assertEquals(Collection<?>, Collection<?>)]https://jitpack.io/com/github/cbeust/testng/master-6.12-g8a3b74b-30/javadoc/org/testng/Assert.html#assertEquals-java.util.Collection-java.util.Collection-l) is being called instead of the Iterable version banned by modernizer. The question is why.

Not having this ambiguity seems like a place where JUnit 5 would be better. But that's a later problem.

On a related note, the Collection version has the same behavior as the Iterable version. Should it be banned too?
https://github.com/cbeust/testng/blob/master/src/main/java/org/testng/Assert.java#L809
https://github.com/cbeust/testng/blob/master/src/main/java/org/testng/Assert.java#L855

Reply:
Also: is modernizer documented anywhere? There are no examples of banning an entire class in our violations.xml file, and I have neither guessed the syntax nor been able to divine it from the source code.

Reply:
It's documented here: https://github.com/andrewgaul/modernizer-maven-plugin

Reply:
And using illegalClasses doesn't work for static importing methods in a class:

https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java#L197

What a rabbit hole this seemingly simple change turned into...

Reply:
It looks like Modernizer only works for exact matches, so we'll need to use `illegalClasses` for this.

Reply:
> It's documented here: https://github.com/andrewgaul/modernizer-maven-plugin

Not the syntax for the violations.xml file. And it isn't in the linked ModernizerMojo class either:

https://github.com/andrewgaul/modernizer-maven-plugin/blob/master/src/main/java/org/gaul/modernizer_maven_plugin/ModernizerMojo.java